src/BTVNanoCommissioning/workflows/ttdilep_valid_sf.py

```diff
@@ -4,155 +4,91 @@
-            "ntbjet": hist.Hist("Counts", dataset_axis, ntbjet_axis),
-            "nmu": hist.Hist("Counts", dataset_axis, nmu_axis),
-            "lmupt": hist.Hist("Counts", dataset_axis, lmupt_axis),
-            "ljpt": hist.Hist("Counts", dataset_axis, flav_axis, ljpt_axis),
-            "sljpt": hist.Hist("Counts", dataset_axis, flav_axis, sljpt_axis),
-            "ljeta": hist.Hist("Counts", dataset_axis, flav_axis, ljeta_axis),
-            "sljeta": hist.Hist("Counts", dataset_axis, flav_axis, sljeta_axis),
-            "ljdr": hist.Hist("Counts", dataset_axis, flav_axis, ljdr_axis),
-            "sljdr": hist.Hist("Counts", dataset_axis, flav_axis, sljdr_axis),
+            self._pu = load_pu(self._campaign, correction_config[self._campaign]["PU"])
+        if isJERC:
+            self._jet_factory = load_jetfactory(
+                self._campaign, correction_config[self._campaign]["JME"]
+            )
+        _hist_event_dict = histogrammer("ttdilep_sf")
+        self.make_output = lambda: {
+            "sumw": processor.defaultdict_accumulator(float),
+            **_hist_event_dict,
         }
 
-        self.sf_dict = list(_hist_sf_dict.keys())
-        self.event_hists = list(_hist_event_dict.keys())
-        self.btagDeephists = list(_hist_btagDeepdict.keys())
-        _hist_dict = {**_hist_sf_dict, **_hist_event_dict, **_hist_btagDeepdict}
-        # }
-        self._accumulator = processor.dict_accumulator(_hist_dict)
-        self._accumulator["sumw"] = processor.defaultdict_accumulator(float)
-        ## Load corrections
-        (
-            self._deepcsvb_sf,
-            self._deepcsvc_sf,
-            self._deepjetb_sf,
-            self._deepjetc_sf,
-        ) = load_BTV(self._campaign, correction_config[self._campaign]["BTV"])
-        self._pu = load_pu(self._campaign, correction_config[self._campaign]["PU"])
-
     @property
     def accumulator(self):
         return self._accumulator
 
     def process(self, events):
-        output = self.accumulator.identity()
+        output = self.make_output()
         dataset = events.metadata["dataset"]
         isRealData = not hasattr(events, "genWeight")
-
         if isRealData:
-            output["sumw"][dataset] += 1.0
+            output["sumw"] = len(events)
         else:
-            output["sumw"][dataset] += ak.sum(events.genWeight)
+            output["sumw"] = ak.sum(events.genWeight)
+            if self.isJERC:
+                events.Jet = self._jet_factory["mc"].build(
+                    add_jec_variables(events.Jet, events.fixedGridRhoFastjetAll),
+                    lazy_cache=events.caches[0],
+                )
         req_lumi = np.ones(len(events), dtype="bool")
-        if isRealData:
-            req_lumi = lumiMasks[self._year](events.run, events.luminosityBlock)
+        # if isRealData:
+        #     req_lumi = lumiMasks[self._year](events.run, events.luminosityBlock)
```

Comment:
why is this commented out?

Reply:
I make some test for run3 data, temporary comment out( would uncomment it back)